_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+      const keys = this.searchable[k];
+      if (this.matchQuery(search, keys)) {
+        return keys;
+      }
+    }
+    return [];
+  },
+
+  get hashKeys() {
+    const hash = this.hash;
+    for (var k in this.hashable) {
+      const keys = this.hashable[k];
+      if (this.matchQuery(hash, keys)) {
+        return keys;
+      }
+    }
+    return [];
+  },
+
+  get isHash() {
+    return !!this.hashKeys.length;
+  },
+
+  /*
+   * Search Keys
+   */
+
+  get edit() {
+    return this.state.edit;
+  },
+  set edit(_e) {
+    const e = parseInt(_e, 10);
+    this.state.edit = modulo(e, 2);
+  },
+
+
+  /*
+   * Control keys
+   */
+
+  get token() {
+    const username = 'john_hoffer@hms.harvard.edu';
+    const password = document.minerva_password;
+    const pass = new Promise(function(resolve, reject) {
+      if (password != undefined) {
+        resolve(password);
+      }
+      const selector = '#password_modal';
+      $(selector).modal('show');
+      $(selector).find('form').submit(function(e){
+        $(selector).find('form').off();
+        $(this).closest('.modal').modal('hide');
+        const formData = parseForm(e.target);
+       
+        // Get password from form
+        const p = formData.p;
+        document.minerva_password = p;
+        resolve(p);
+        return false;
+      });
+    });
+    return authenticate(username, pass);
+  },
+
+  get editing() {
+    if (this.edit) {
+      return this.state.editing;
+    }
+    return 0;
+  },
+  set editing(_e) {
+    const e = parseInt(_e, 10);
+    this.state.editing = modulo(e, 3);
+    this.newView(true);
+  },
+
+  get drawing() {
+    return this.state.drawing;
+  },
+  set drawing(_d) {
+    const d = parseInt(_d, 10);
+    this.state.drawing = modulo(d, 3);
+    this.newView(true);
+  },
+
+  get mouseXY() {
+    const e = this.state.mouseEvent;
+    const pos = OpenSeadragon.getMousePosition(e);
+    return this.normalize(pos);
+  },
+  set mouseXY(e) {
+    this.state.mouseEvent = e;
+  },
+
+  /*
+   * Hash Keys
+   */
+
+  get v() {
+    return this.state.v;
+  },
+  set v(_v) {
+    const viewer = this.viewer;
+    this.state.v = _v.map(parseFloat);
+  },
+
+  get g() {
+    const g = this.state.g;
+    const count = this.cgs.length;
+    return g < count ? g : 0;
+  },
+  set g(_g) {
+    const g = parseInt(_g, 10);
+    const count = this.cgs.length;
+    this.state.g = modulo(g, count);
+  },
+
+  /*
+   * Exhibit Hash Keys
+   */
+
+  get w() {
+    const w = this.state.w[this.s] || 0;
+    const count = this.waypoints.length;
+    return w < count ? w : 0;
+  },
+
+  set w(_w) {
+    const w = parseInt(_w, 10);
+    const count = this.waypoints.length;
+    this.state.w[this.s] = modulo(w, count);
+
+    // Set group, viewport from waypoint
+    const waypoint = this.waypoint;
+
+    this.slower();
+    this.g = gFromWaypoint(waypoint, this.cgs);
+    this.v = vFromWaypoint(waypoint);
+    this.o = oFromWaypoint(waypoint);
```

Comment:
Setting the waypoint automatically applies the waypoint's group index, viewport coordinates, and overlay position.